src/common/file_system.cpp

```diff
@@ -227,6 +228,32 @@ time_t FileSystem::GetLastModifiedTime(FileHandle &handle) {
 	return s.st_mtime;
 }
 
+uint8_t FileSystem::GetFileType(FileHandle &handle) {
```

Comment:
Can this return FileType directly instead of uint8_t?

Reply:
Probably, I've just followed existing practice for `FileFlags` and such.

Reply:
FileFlags does it this way because the options can be combined together, the FileType should be mutually exclusive right? (i.e. a file can’t be both a pipe and a directory)

Reply:
Yep, they are exclusive.